diff -ruwp gcc/config/avr/avr.c gcc/config/avr/avr.c
--- gcc/config/avr/avr.c	2010-08-30 11:06:12.000000000 -0500
+++ gcc/config/avr/avr.c	2010-08-30 11:10:04.000000000 -0500
@@ -1029,7 +1029,8 @@ expand_prologue (void)
 		}
 	      else if ((!AVR_XMEGA && TARGET_NO_INTERRUPTS)
 		       || (!AVR_XMEGA && cfun->machine->is_signal)
-		       || (!AVR_XMEGA && cfun->machine->is_OS_main))
+		       || (!AVR_XMEGA && cfun->machine->is_OS_main)
+		       || (AVR_XMEGA && cfun->machine->is_nmi))
 		{
 		  insn = 
 		    emit_insn (gen_movhi_sp_r_irq_off (stack_pointer_rtx, 
@@ -1202,7 +1203,8 @@ expand_epilogue (void)
 		  emit_move_insn (stack_pointer_rtx, frame_pointer_rtx);
 		}
 	      else if ((!AVR_XMEGA && TARGET_NO_INTERRUPTS) 
-		       || (!AVR_XMEGA && cfun->machine->is_signal))
+		       || (!AVR_XMEGA && cfun->machine->is_signal)
+		       || (AVR_XMEGA && cfun->machine->is_nmi))
 		{
 		  emit_insn (gen_movhi_sp_r_irq_off (stack_pointer_rtx, 
 						     frame_pointer_rtx));
diff -ruwp gcc/function.c gcc/function.c
--- gcc/function.c	2009-11-13 13:57:51.000000000 -0600
+++ gcc/function.c	2010-08-30 11:07:40.000000000 -0500
@@ -4936,6 +4936,14 @@ contains (const_rtx insn, VEC(int,heap) 
 }
 
 int
+prologue_contains (const_rtx insn)
+{
+  if (contains (insn, &prologue))
+    return 1;
+  return 0;
+}
+
+int
 prologue_epilogue_contains (const_rtx insn)
 {
   if (contains (insn, &prologue))
diff -ruwp gcc/rtl.h gcc/rtl.h
--- gcc/rtl.h	2009-04-27 06:55:13.000000000 -0500
+++ gcc/rtl.h	2010-08-30 11:07:40.000000000 -0500
@@ -2181,6 +2181,7 @@ extern void print_inline_rtx (FILE *, co
 
 /* In function.c */
 extern void reposition_prologue_and_epilogue_notes (void);
+extern int prologue_contains (const_rtx);
 extern int prologue_epilogue_contains (const_rtx);
 extern int sibcall_epilogue_contains (const_rtx);
 extern void mark_temp_addr_taken (rtx);
