Index: gcc-4_1-branch/gcc/config/avr/avr.c
===================================================================
--- gcc-4_1-branch/gcc/config/avr/avr.c
+++ gcc-4_1-branch/gcc/config/avr/avr.c
@@ -4679,7 +4710,7 @@
     }
   else
     {
-      const char *func_name = IDENTIFIER_POINTER (DECL_NAME (*node));
+      const char *func_name = IDENTIFIER_POINTER (DECL_ASSEMBLER_NAME (*node));
       const char *attr = IDENTIFIER_POINTER (name);
 
       /* If the function has the 'signal' or 'interrupt' attribute, test to
