--- gcc/ada/gnat1drv.adb.orig	Tue Apr 26 23:00:00 2005
+++ gcc/ada/gnat1drv.adb	Tue Apr 26 23:00:15 2005
@@ -192,6 +192,42 @@
          Write_Str (Gnat_Version_String);
          Write_Str (" Copyright 1992-2004 Free Software Foundation, Inc.");
          Write_Eol;
+         Write_Eol;
+         Write_Str ("Source Search Path:");
+         Write_Eol;
+
+         for J in 1 .. Nb_Dir_In_Src_Search_Path loop
+            Write_Str ("   ");
+
+            if Dir_In_Src_Search_Path (J)'Length = 0 then
+               Write_Str ("<Current_Directory>");
+            else
+               Write_Str (To_Host_Dir_Spec
+                          (Dir_In_Src_Search_Path (J).all, True).all);
+            end if;
+
+            Write_Eol;
+         end loop;
+
+         Write_Eol;
+         Write_Eol;
+         Write_Str ("Object Search Path:");
+         Write_Eol;
+
+         for J in 1 .. Nb_Dir_In_Obj_Search_Path loop
+            Write_Str ("   ");
+
+            if Dir_In_Obj_Search_Path (J)'Length = 0 then
+               Write_Str ("<Current_Directory>");
+            else
+               Write_Str (To_Host_Dir_Spec
+                          (Dir_In_Obj_Search_Path (J).all, True).all);
+            end if;
+
+            Write_Eol;
+         end loop;
+
+         Write_Eol;
       end if;
 
       --  Before we do anything else, adjust certain global values for
