diff -Naurp binutils/size.c binutils/size.c
--- binutils/size.c	2011-09-05 15:59:51.000000000 +0300
+++ binutils/size.c	2011-09-05 16:03:16.000000000 +0300
@@ -120,6 +120,7 @@ typedef struct
 avr_device_t avr[] =
 {
 	{"atxmega256a3",  AVR264K, AVR16K, AVR4K},
+	{"atxmega256a3u", AVR264K, AVR16K, AVR4K},
 	{"atxmega256a3b", AVR264K, AVR16K, AVR4K},
 	{"atxmega256a3bu",AVR264K, AVR16K, AVR4K},
 	{"atxmega256d3",  AVR264K, AVR16K, AVR4K},
@@ -128,11 +129,13 @@ avr_device_t avr[] =
 	{"atmega2561",    AVR256K, AVR8K,  AVR4K},
 
 	{"atxmega192a3",  AVR200K, AVR16K, AVR2K},
+	{"atxmega192a3u", AVR200K, AVR16K, AVR2K},
 	{"atxmega192d3",  AVR200K, AVR16K, AVR2K},
 
 	{"atxmega128a1",  AVR136K, AVR8K,  AVR2K},
 	{"atxmega128a1u", AVR136K, AVR8K,  AVR2K},
 	{"atxmega128a3",  AVR136K, AVR8K,  AVR2K},
+	{"atxmega128a3u", AVR136K, AVR8K,  AVR2K},
 	{"atxmega128b1",  AVR136K, AVR8K,  AVR2K},
 	{"atxmega128d3",  AVR136K, AVR8K,  AVR2K},
 
@@ -150,6 +153,7 @@ avr_device_t avr[] =
 	{"atxmega64a1",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64a1u",  AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64a3",   AVR68K,  AVR4K,  AVR2K},
+	{"atxmega64a3u",  AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64d3",   AVR68K,  AVR4K,  AVR2K},
 
 	{"at90can64",     AVR64K,  AVR4K,  AVR2K},
diff -Naurp gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2011-09-05 15:59:51.000000000 +0300
+++ gas/config/tc-avr.c	2011-09-05 16:03:16.000000000 +0300
@@ -263,15 +263,19 @@ static struct mcu_type_s mcu_types[] =
   {"atxmega32d4", AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
   {"atxmega32x1", AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
   {"atxmega64a3", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
+  {"atxmega64a3u",AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64d3", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64a1", AVR_ISA_XMEGA,  bfd_mach_avrxmega5},
   {"atxmega64a1u",AVR_ISA_XMEGA,  bfd_mach_avrxmega5},
   {"atxmega128a3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega128a3u",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128b1", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega192a3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega192a3u",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega192d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega256a3u",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3b",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3bu",AVR_ISA_XMEGA,bfd_mach_avrxmega6},
   {"atxmega256d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
diff -Naurp gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2011-09-05 15:59:51.000000000 +0300
+++ gas/doc/c-avr.texi	2011-09-05 16:03:16.000000000 +0300
@@ -93,16 +93,17 @@ Instruction set avrxmega3 is for the XME
 memory space and greater than 64K data space (MCU types: none).
 
 Instruction set avrxmega4 is for the XMEGA AVR core with up to 64K program
-memory space and less than 64K data space (MCU types: atxmega64a3, atxmega64d3).
+memory space and less than 64K data space (MCU types: atxmega64a3, atxmega64a3u,
+atxmega64d3).
 
 Instruction set avrxmega5 is for the XMEGA AVR core with up to 64K program 
 memory space and greater than 64K data space (MCU types: atxmega64a1,
 atxmega64a1u).
 
 Instruction set avrxmega6 is for the XMEGA AVR core with up to 256K program
-memory space and less than 64K data space (MCU types: atxmega128a3,
-atxmega128d3, atxmega192a3, atxmega128b1, atxmega192d3, atxmega256a3, 
-atxmega256a3b, atxmega256a3bu, atxmega192d3).
+memory space and less than 64K data space (MCU types: atxmega128a3, atxmega128a3u,
+atxmega128d3, atxmega192a3, atxmega192a3u, atxmega128b1, atxmega192d3, atxmega256a3,
+atxmega256a3u, atxmega256a3b, atxmega256a3bu, atxmega192d3).
 
 Instruction set avrxmega7 is for the XMEGA AVR core with up to 256K program
 memory space and greater than 64K data space (MCU types: atxmega128a1,
