diff -Naurp binutils/size.c binutils/size.c
--- binutils/size.c	2011-06-24 16:23:14.000000000 +0530
+++ binutils/size.c	2011-06-24 10:48:45.000000000 +0530
@@ -119,6 +119,9 @@ typedef struct
 
 avr_device_t avr[] =
 {
+	{"atmega256rfa2", AVR256K, AVR32K, AVR8K},
+	{"atmega256rfr2", AVR256K, AVR32K, AVR8K},
+
 	{"atxmega256a3",  AVR264K, AVR16K, AVR4K},
 	{"atxmega256a3u", AVR264K, AVR16K, AVR4K},
 	{"atxmega256a3b", AVR264K, AVR16K, AVR4K},
@@ -132,6 +135,9 @@ avr_device_t avr[] =
 	{"atxmega192a3u", AVR200K, AVR16K, AVR2K},
 	{"atxmega192d3",  AVR200K, AVR16K, AVR2K},
 
+	{"atmega128rfa2", AVR128K, AVR16K, AVR4K},
+	{"atmega128rfr2", AVR128K, AVR16K, AVR4K},
+
 	{"atxmega128a1",  AVR136K, AVR8K,  AVR2K},
 	{"atxmega128a1u", AVR136K, AVR8K,  AVR2K},
 	{"atxmega128a3",  AVR136K, AVR8K,  AVR2K},
@@ -155,6 +161,9 @@ avr_device_t avr[] =
 	{"atxmega64a3",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64a3u",  AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64d3",   AVR68K,  AVR4K,  AVR2K},
+	
+	{"atmega64rfa2",  AVR64K,  AVR8K,  AVR2K},
+	{"atmega64rfr2",  AVR64K,  AVR8K,  AVR2K},
 
 	{"at90can64",     AVR64K,  AVR4K,  AVR2K},
 	{"at90scr100",    AVR64K,  AVR4K,  AVR2K},
diff -Naurp gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2011-06-24 16:23:14.000000000 +0530
+++ gas/config/tc-avr.c	2011-06-24 16:08:14.000000000 +0530
@@ -203,6 +203,8 @@ static struct mcu_type_s mcu_types[] =
   {"atmega3290p",AVR_ISA_AVR5,    bfd_mach_avr5},
   {"atmega3290pa",AVR_ISA_AVR5,   bfd_mach_avr5},
   {"atmega406",  AVR_ISA_AVR5,    bfd_mach_avr5},
+  {"atmega64rfa2", AVR_ISA_AVR5,  bfd_mach_avr5},
+  {"atmega64rfr2", AVR_ISA_AVR5,  bfd_mach_avr5},
   {"atmega64",   AVR_ISA_AVR5,    bfd_mach_avr5},
   {"atmega640",  AVR_ISA_AVR5,    bfd_mach_avr5},
   {"atmega644",  AVR_ISA_AVR5,    bfd_mach_avr5},
@@ -250,11 +252,15 @@ static struct mcu_type_s mcu_types[] =
   {"atmega1281", AVR_ISA_AVR51,   bfd_mach_avr51},
   {"atmega1284p",AVR_ISA_AVR51,   bfd_mach_avr51},
   {"atmega128rfa1",AVR_ISA_AVR51, bfd_mach_avr51},
+  {"atmega128rfa2",AVR_ISA_AVR51, bfd_mach_avr51},
+  {"atmega128rfr2",AVR_ISA_AVR51, bfd_mach_avr51},
   {"at90can128", AVR_ISA_AVR51,   bfd_mach_avr51},
   {"at90usb1286",AVR_ISA_AVR51,   bfd_mach_avr51},
   {"at90usb1287",AVR_ISA_AVR51,   bfd_mach_avr51},
   {"atmega2560", AVR_ISA_AVR6,    bfd_mach_avr6},
   {"atmega2561", AVR_ISA_AVR6,    bfd_mach_avr6},
+  {"atmega256rfa2", AVR_ISA_AVR6, bfd_mach_avr6},
+  {"atmega256rfr2", AVR_ISA_AVR6, bfd_mach_avr6},
   {"atxmega16a4", AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
   {"atxmega16a4u",AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
   {"atxmega16d4", AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
diff -Naurp gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2011-06-24 16:23:14.000000000 +0530
+++ gas/doc/c-avr.texi	2011-06-24 16:19:20.000000000 +0530
@@ -68,22 +68,23 @@ memory space (MCU types: at90pwm161, atm
 atmega163, atmega164a, atmega164p, atmega165, atmega165a, atmega165p, atmega168, 
 atmega168a, atmega168p, atmega169, atmega169p, atmega169pa,
 atmega32, atmega323, atmega324a, atmega324p, atmega324pa, atmega325, atmega325a,
-atmega325p, atmega325pa, atmega3250, atmega3250a, atmega3250p, atmega3250pa, atmega328, atmega328p,
-atmega329, atmega329a, atmega329p, atmega329pa, atmega3290, atmega3290a,
-atmega3290p, atmega3290pa, atmega406, atmega64, atmega640, atmega644, atmega644a, atmega644p,
-atmega644pa, atmega645, atmega645a, atmega645p, atmega6450, atmega6450a,
-atmega6450p, atmega649, atmega649a, atmega649p, atmega6490, atmega6490a,
-atmega6490p, atmega64hve, atmega16hva, atmega16hva2, atmega16hvb, atmega32hvb,
-atmega16hvbrevb, atmega32hvbrevb, at90can32, at90can64, at90pwm216, at90pwm316,
-atmega16u4, atmega32c1, atmega64c1, atmega64m1, atmega16m1, atmega32m1, atmega64m1,
+atmega325p, atmega325pa, atmega3250, atmega3250a, atmega3250p, atmega3250pa, 
+atmega328, atmega328p,atmega329, atmega329a, atmega329p, atmega329pa, atmega3290, 
+atmega3290a, atmega3290p, atmega3290pa, atmega406,  atmega64rfa2, atmega64rfr2, 
+atmega64, atmega640, atmega644, atmega644a, atmega644p, atmega644pa, atmega645, 
+atmega645a, atmega645p, atmega6450, atmega6450a, atmega6450p, atmega649, 
+atmega649a, atmega649p, atmega6490, atmega6490a, atmega6490p, atmega64hve, 
+atmega16hva, atmega16hva2, atmega16hvb, atmega32hvb, atmega16hvbrevb, 
+atmega32hvbrevb, at90can32, at90can64, at90pwm216, at90pwm316, atmega16u4, 
+atmega32c1, atmega64c1, atmega64m1, atmega16m1, atmega32m1, atmega64m1,
 atmega16u4, atmega32u4, atmega32u6, at90usb646, at90usb647, at94k, at90scr100).
 
 Instruction set avr51 is for the enhanced AVR core with exactly 128K program
 memory space (MCU types: atmega128, atmega1280, atmega1281, atmega1284p,
-atmega128rfa1, at90can128, at90usb1286, at90usb1287, m3000).
+atmega128rfa1, atmega128rfa2, atmega128rfr2, at90can128, at90usb1286, at90usb1287, m3000).
 
 Instruction set avr6 is for the enhanced AVR core with a 3-byte PC (MCU types:
-atmega2560, atmega2561).
+atmega2560, atmega2561, atmega256rfa2, atmega256rfr2).
 
 Instruction set avrxmega2 is for the XMEGA AVR core with 8K to 64K program
 memory space and less than 64K data space (MCU types: atxmega16a4, atxmega16a4u,
