diff -Naurp binutils/size.c binutils/size.c
--- binutils/size.c	2011-09-05 16:08:16.000000000 +0300
+++ binutils/size.c	2011-09-05 16:08:24.000000000 +0300
@@ -108,6 +108,7 @@ static int return_code = 0;
 #define AVR200K 204800UL
 #define AVR256K 262144UL
 #define AVR264K 270336UL
+#define AVR392K 401408UL
 
 typedef struct
 {
@@ -119,6 +120,8 @@ typedef struct
 
 avr_device_t avr[] =
 {
+ 	{"atxmega384c3",  AVR392K, AVR32K, AVR4K},
+ 
 	{"atmega256rfa2", AVR256K, AVR32K, AVR8K},
 	{"atmega256rfr2", AVR256K, AVR32K, AVR8K},
 
diff -Naurp gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2011-09-05 16:08:16.000000000 +0300
+++ gas/config/tc-avr.c	2011-09-05 16:08:24.000000000 +0300
@@ -286,6 +286,7 @@ static struct mcu_type_s mcu_types[] =
   {"atxmega256a3b",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3bu",AVR_ISA_XMEGA,bfd_mach_avrxmega6},
   {"atxmega256d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega384c3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128a1", AVR_ISA_XMEGA, bfd_mach_avrxmega7},
   {"atxmega128a1u",AVR_ISA_XMEGA, bfd_mach_avrxmega7},
   {"attiny4",      AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
diff -Naurp gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2011-09-05 16:08:16.000000000 +0300
+++ gas/doc/c-avr.texi	2011-09-05 16:08:24.000000000 +0300
@@ -104,9 +104,9 @@ atxmega64a1u).
 Instruction set avrxmega6 is for the XMEGA AVR core with up to 256K program
 memory space and less than 64K data space (MCU types: atxmega128a3, atxmega128a3u,
 atxmega128d3, atxmega192a3, atxmega192a3u, atxmega128b1, atxmega192d3, atxmega256a3,
-atxmega256a3u, atxmega256a3b, atxmega256a3bu, atxmega192d3).
+atxmega256a3u, atxmega256a3b, atxmega256a3bu, atxmega192d3, atxmega384c3).
 
-Instruction set avrxmega7 is for the XMEGA AVR core with up to 256K program
+Instruction set avrxmega7 is for the XMEGA AVR core with greater than 64K program
 memory space and greater than 64K data space (MCU types: atxmega128a1,
 atxmega128a1u).
 
