diff -Naurp binutils/size.c binutils/size.c
--- binutils/size.c	2011-09-07 18:04:50.000000000 +0300
+++ binutils/size.c	2011-09-07 18:11:50.000000000 +0300
@@ -165,6 +165,7 @@ avr_device_t avr[] =
 	{"atxmega64a1u",  AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64a3",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64a3u",  AVR68K,  AVR4K,  AVR2K},
+	{"atxmega64b1",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64b3",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64d3",   AVR68K,  AVR4K,  AVR2K},
 	{"atxmega64d4",   AVR68K,  AVR4K,  AVR2K},
diff -Naurp gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2011-09-07 18:04:50.000000000 +0300
+++ gas/config/tc-avr.c	2011-09-07 18:10:28.000000000 +0300
@@ -276,6 +276,7 @@ static struct mcu_type_s mcu_types[] =
   {"atxmega32x1", AVR_ISA_XMEGA,  bfd_mach_avrxmega2},
   {"atxmega64a3", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64a3u",AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
+  {"atxmega64b1", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64b3", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64d3", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
   {"atxmega64d4", AVR_ISA_XMEGA,  bfd_mach_avrxmega4},
diff -Naurp gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2011-09-07 18:04:50.000000000 +0300
+++ gas/doc/c-avr.texi	2011-09-07 18:12:34.000000000 +0300
@@ -95,7 +95,7 @@ memory space and greater than 64K data s
 
 Instruction set avrxmega4 is for the XMEGA AVR core with up to 64K program
 memory space and less than 64K data space (MCU types: atxmega64a3, atxmega64a3u,
-atxmega64b3, atxmega64d3, atxmega64d4).
+atxmega64b1, atxmega64b3, atxmega64d3, atxmega64d4).
 
 Instruction set avrxmega5 is for the XMEGA AVR core with up to 64K program 
 memory space and greater than 64K data space (MCU types: atxmega64a1,
