diff -Naurp binutils/size.c binutils/size.c
--- binutils/size.c	2011-09-15 11:59:56.000000000 +0300
+++ binutils/size.c	2011-09-15 12:04:20.000000000 +0300
@@ -149,6 +149,7 @@ avr_device_t avr[] =
 	{"atxmega128b1",  AVR136K, AVR8K,  AVR2K},
 	{"atxmega128b3",  AVR136K, AVR8K,  AVR2K},
 	{"atxmega128d3",  AVR136K, AVR8K,  AVR2K},
+	{"atxmega128d4",  AVR136K, AVR8K,  AVR2K},
 
 	{"at43usb320",    AVR128K, 608UL,  0UL},
 	{"at90can128",    AVR128K, AVR4K,  AVR4K},
diff -Naurp gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2011-09-15 11:59:56.000000000 +0300
+++ gas/config/tc-avr.c	2011-09-15 12:04:20.000000000 +0300
@@ -291,6 +291,7 @@ static struct mcu_type_s mcu_types[] =
   {"atxmega128b1", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128b3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega128d4", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega192a3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega192a3u",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega192d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
diff -Naurp gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2011-09-15 11:59:56.000000000 +0300
+++ gas/doc/c-avr.texi	2011-09-15 12:04:20.000000000 +0300
@@ -103,8 +103,9 @@ atxmega64a1u).
 
 Instruction set avrxmega6 is for the XMEGA AVR core with up to 256K program
 memory space and less than 64K data space (MCU types: atxmega128a3, atxmega128a3u,
-atxmega128d3, atxmega192a3, atxmega192a3u, atxmega128b1, atxmega128b3, atxmega192d3,
-atxmega256a3,atxmega256a3u, atxmega256a3b, atxmega256a3bu, atxmega192d3, atxmega384c3).
+atxmega128d3, atxmega128d4, atxmega192a3, atxmega192a3u, atxmega128b1, atxmega128b3, 
+atxmega192d3, atxmega256a3,atxmega256a3u, atxmega256a3b, atxmega256a3bu, atxmega192d3, 
+atxmega384c3).
 
 Instruction set avrxmega7 is for the XMEGA AVR core with greater than 64K program
 memory space and greater than 64K data space (MCU types: atxmega128a1,
