diff -u -r gcc-old/gcc/ada/Make-lang.in gcc-new/gcc/ada/Make-lang.in
--- gcc-old/gcc/ada/Make-lang.in	2004-01-31 01:45:25.000000000 +0100
+++ gcc-new/gcc/ada/Make-lang.in	2004-03-19 14:40:17.000000000 +0100
@@ -1229,7 +1229,7 @@
    $(TREE_H) flags.h expr.h convert.h defaults.h ada/ada.h ada/types.h \
    ada/atree.h ada/nlists.h ada/elists.h ada/sinfo.h ada/einfo.h ada/namet.h \
    ada/stringt.h ada/uintp.h ada/fe.h $(ADA_TREE_H) ada/gigi.h gt-ada-utils.h \
-   gtype-ada.h
+   gtype-ada.h target.h
 
 ada/utils2.o : ada/utils2.c $(CONFIG_H) $(SYSTEM_H) coretypes.h $(TM_H) \
    $(TREE_H) flags.h ada/ada.h ada/types.h ada/atree.h ada/nlists.h \
diff -u -r gcc-old/gcc/ada/utils.c gcc-new/gcc/ada/utils.c
--- gcc-old/gcc/ada/utils.c	2003-11-24 18:18:19.000000000 +0100
+++ gcc-new/gcc/ada/utils.c	2004-03-19 14:40:17.000000000 +0100
@@ -36,6 +36,7 @@
 #include "ggc.h"
 #include "debug.h"
 #include "convert.h"
+#include "target.h"
 
 #include "ada.h"
 #include "types.h"
@@ -1566,15 +1567,16 @@
 	break;
 
       case ATTR_LINK_SECTION:
-#ifdef ASM_OUTPUT_SECTION_NAME
-	DECL_SECTION_NAME (decl)
-	  = build_string (IDENTIFIER_LENGTH (attr_list->name),
-			  IDENTIFIER_POINTER (attr_list->name));
-	DECL_COMMON (decl) = 0;
-#else
-	post_error ("?section attributes are not supported for this target",
-		    attr_list->error_point);
-#endif
+	if (targetm.have_named_sections)
+	  {
+	    DECL_SECTION_NAME (decl)
+	      = build_string (IDENTIFIER_LENGTH (attr_list->name),
+			      IDENTIFIER_POINTER (attr_list->name));
+	    DECL_COMMON (decl) = 0;
+	  }
+	else
+	  post_error ("?section attributes are not supported for this target",
+		      attr_list->error_point);
 	break;
       }
 }
