--- libiberty/Makefile.in.orig	2010-03-02 06:09:56 +0000
+++ libiberty/Makefile.in	2010-09-05 20:53:58 +0000
@@ -321,7 +321,8 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+#install: install_to_$(INSTALL_DEST) install-subdir
+install:
 
 # This is tricky.  Even though CC in the Makefile contains
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
